### ISSUE TITLE
Couldn't read result file.

### ISSUE THREAD

**Issue by jingydz** [fangzhonghai/HLA-HD#2]
Hi,
    When I finish the HLA-HD pipeline, I got something like: >sample ID
A       HLA-A*24:02:01  HLA-A*02:01:01
B       HLA-B*39:01:01  HLA-B*40:01:02
C       HLA-C*07:02:01  -
DRB1    HLA-DRB1*11:01:01       HLA-DRB1*08:03:02
DQA1    HLA-DQA1*01:03:01       HLA-DQA1*05:05:01
DQB1    HLA-DQB1*06:01:01       HLA-DQB1*03:01:01
DPA1    HLA-DPA1*02:02:02       -
DPB1    HLA-DPB1*135:01 HLA-DPB1*02:02:01
DMA     HLA-DMA*01:02   HLA-DMA*01:01:01
DMB     HLA-DMB*01:01:01        -
DOA     HLA-DOA*01:01:01        HLA-DOA*01:01:04
DOB     HLA-DOB*01:01:01        -
DRA     HLA-DRA*01:01:01        HLA-DRA*01:02:02
DRB2    HLA-DRB2*01:01  -
DRB3    HLA-DRB3*02:02:01       -
DRB4    Not typed       Not typed
DRB5    Not typed       Not typed
DRB6    Not typed       Not typed
DRB7    Not typed       Not typed
DRB8    Not typed       Not typed
DRB9    HLA-DRB9*01:01:01       HLA-DRB9*01:02:01
DPA2    HLA-DPA2*02:01  HLA-DPA2*01:01:02
E       HLA-E*01:03:01  HLA-E*01:03:02
F       HLA-F*01:01:01  -
G       HLA-G*01:04:01  HLA-G*01:01:01
Couldn't read result file.
J       HLA-J*01:01:01  -
K       HLA-K*01:02     -
L       HLA-L*01:02     HLA-L*01:01:01
T       HLA-T*01:01:01  -
V       HLA-V*01:01:01  -
W       HLA-W*03:01:01  HLA-W*01:01:01
Y       Not typed       Not typed
Do you know why? If you can reply to me I would really appreciate it.

**Comment by fangzhonghai** [comment]
It seems that something wrong happened when performed "pick_up_allele H" in $sampleID/pickup.sh. Is it ok to use the original software (hlahd.1.2.1/bin/hlahd.sh)?

**Comment by jingydz** [comment]
The version I use is hlahd.1.3.0. I don't know what's wrong. 
And sample1_R1(R2).H_exonX.sam files are created in the sample1/exon directory, but some files seem to have only headers in them.
Similarly, in the intron directory, there is sample1_R1(R2).h_intronX.sam, but it only has 2*7 files.
But in the result directory, there are no sample1_H.est.txt or sample1_H.read.txt files.

**Comment by fangzhonghai** [comment]
You could use hlahd.1.3.0/bin/hlahd.sh for your job. If exception happens as before, I think hlahd.1.3.0 itself has bugs.

**Comment by jingydz** [comment]
Yes, the sh file I'm using is hlahd.sh version 1.3.0. I ran it again several times and got the same result. And Couldn't read result file appeared where HLA-H* should have appeared. I tried to understand his source code, but I failed. (I'm a senior now, so it's kind of hard for me.) 
`    in.open(rname.c_str(),ios::in);
    if(in){
      while(getline(in,sline)){
        n = 0;
        word = GetWordtab(n,sline);
        if(word == "No candidate."){
          pcnt = 1;
          amgcnt = 1;
          for(i=0;i<2;i++){
            pairs[i] = new string[pcnt];
          }
          pairs[0][0] = "NULL";
        }
        else if(word == "#Best allele pair"){
          word = GetWordtab(n,sline);
          pcnt = atoi(word.c_str());      
          for(i=0;i<2;i++){
            pairs[i] = new string[pcnt];
          }
          for(i=0;i<pcnt;i++){
            getline(in,sline);
            n = 0;
            for(j=0;j<2;j++){
              word = GetWordtab(n,sline);
              if(word.size() > 0){
                pairs[j][i] = word;
              }
              else{
                pairs[j][i] = "-";
              }
            }
          }
        }
      }
    }
    else{
      cout << "Couldn't read result file." << endl;
      exit(0);
    }`

**Comment by jingydz** [comment]
[pick_up_allele.cpp2txt.txt](https://github.com/fangzhonghai/HLA-HD/files/5862395/pick_up_allele.cpp2txt.txt)

**Comment by fangzhonghai** [comment]
> Yes, the sh file I'm using is hlahd.sh version 1.3.0. I ran it again several times and got the same result. And Couldn't read result file appeared where HLA-H* should have appeared. I tried to understand his source code, but I failed. (I'm a senior now, so it's kind of hard for me.)
> ` in.open(rname.c_str(),ios::in); if(in){ while(getline(in,sline)){ n = 0; word = GetWordtab(n,sline); if(word == "No candidate."){ pcnt = 1; amgcnt = 1; for(i=0;i<2;i++){ pairs[i] = new string[pcnt]; } pairs[0][0] = "NULL"; } else if(word == "#Best allele pair"){ word = GetWordtab(n,sline); pcnt = atoi(word.c_str()); for(i=0;i<2;i++){ pairs[i] = new string[pcnt]; } for(i=0;i<pcnt;i++){ getline(in,sline); n = 0; for(j=0;j<2;j++){ word = GetWordtab(n,sline); if(word.size() > 0){ pairs[j][i] = word; } else{ pairs[j][i] = "-"; } } } } } } else{ cout << "Couldn't read result file." << endl; exit(0); }`

I'm sorry. I don't c++. As author mentioned at https://www.genome.med.kyoto-u.ac.jp/HLA-HD/, version 1.3.0 corrected bug in pm_extract (Avoid bug of “Couldn’t read result file” is occurred in typing result). You could report the issue to author.

**Comment by jingydz** [comment]
ok,  I did it.  Thank you!
